### PR TITLE
Deliver CSP via HTML Meta Tag

### LINKED ISSUE #776: using II for local development in Safari fails
# Motivation

As a developer I want to be able to develop my dapp locally with II installed locally as well. This works well with Chrome or Firefox but unfortunately, fails in Safari.

e.g. given II local path `http://127.0.0.1:8000/?canisterId=rwlgt-iiaaa-aaaaa-aaaaa-cai` I get an error "Fail to load resource through https"

# Reproduction

```
npm init ic
follow the prompt, create a project with II, start dfx and deploy
open the provided II link in Safari
```

# Screenshots

Safari:

Capture d’écran 2022-07-29 à 09 47 41

Chrome for comparison:

Capture d’écran 2022-07-29 à 09 47 52

### PR DESCRIPTION
**Pull request by frederikrothenberger**
In addition to the header, the CSP is now also delivered in the HTML meta tag. This is done because headers are not (yet) certified.
This PR also introduces the feature `II_INSECURE_REQUESTS` which fixes #776.

### CODE REVIEW

File: scripts/build [hunk 1]
```diff
@@ -137,14 +137,21 @@ function build_internet_identity() {
 
   # This enables the "dummy_captcha" feature which makes sure the captcha string
   # is always "a".
-  # WARNING: this MUST be opt-in, because we DO NOT want this in production,
-  # EVAR.
```
Comment (nmattia) [inline scripts/build:141]: beh?

File: src/internet_identity/src/assets.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ lazy_static! {
         format!("sha256-{hash}")
     };
 
-    // The full content of the index.html, after the canister ID (and script tag) have been
+    // The full content of the index.html, after the canister, the script tag and the CSP have been
```
Comment (nmattia) [inline src/internet_identity/src/assets.rs:38]:
```suggestion
    // The full content of the index.html, after the canister ID, the script tag and the CSP have been
```

File: src/internet_identity/src/http.rs [hunk 3]
```diff
@@ -253,6 +214,61 @@ fn security_headers() -> Vec<HeaderField> {
     ]
 }
 
+/// Full content security policy delivered via HTTP response header.
+///
+/// This policy also includes the `frame-ancestors` directive in addition to the policies included in the HTML `meta` tag.
+/// We deliver the CSP by header _and_ meta tag because the headers are not yet certified.
```
Comment (nmattia) [inline src/internet_identity/src/http.rs:220]: ~~Why don't we just deliver it in the meta tag then?~~

Ah, looks like the `meta` tag doesn't support all directives.

File: src/internet_identity/src/http.rs [hunk 4]
```diff
@@ -253,6 +214,61 @@ fn security_headers() -> Vec<HeaderField> {
...
+/// There is currently no other way to allow execution of WebAssembly modules with CSP.
+/// See https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md.
+///
+/// script-src 'unsafe-inline' https: are only there for backwards compatibility and ignored
+/// by modern browsers.
+///
+/// connect-src https://*.ic0.app is required in order for II to be able to fetch the
+/// /.well-known/ii-alternative-origins path of authenticating canisters setting a derivationOrigin.
+///
+/// style-src 'unsafe-inline' is currently required due to the way styles are handled by the
```
Comment (nmattia) [inline src/internet_identity/src/http.rs:243]: I _think_ we can remove this now that we load the css via `link` (for another PR)

**Review by nmattia — APPROVED** [review]
LGTM!

Bunch of code moved and I didn't check line by line what changed but I'll trust you on this

File: src/internet_identity/src/http.rs (hunk 3 above)
Comment (frederikrothenberger) [reply src/internet_identity/src/http.rs:220]: Yes, exactly.

File: scripts/build (hunk 1 above)
Comment (frederikrothenberger) [reply scripts/build:141]: I do not like the `EVAR`... 🙃

**Pull request merged**